solr/core/src/java/org/apache/solr/handler/admin/api/CoreReplicationAPI.java

```diff
@@ -68,6 +71,45 @@ public FileListResponse fetchFileList(
     return doFetchFileList(gen);
   }
 
+  @GET
+  @Path("/files/{filePath}")
+  @Produces({MediaType.TEXT_PLAIN, BINARY_CONTENT_TYPE_V2})
+  public String fetchFile(
```

Comment:
[?] Can this really return a `String`?  Since it can return raw files I would've expected this to have to return `InputStream` or `Object` or something more generic in order to handle the potentially-binary content coming back?